.github/CONTRIBUTING.md

```diff
@@ -1,42 +1,251 @@
+
+We are currently adopting TypeScript as the default language on our projects, the current codebase will be migrated, along the time, from JavaScript to TypeScript.
+
+While we still have a lot of JavaScript files you should not create new ones. As much as possible new code contributions should be in **TypeScript**.
+
+### Blaze vs React
 
-While we still have a lot of CoffeeScript files you should not create new ones. New code contributions should be in **ECMAScript 2015**.
+We are currently adopting React over Blaze as our UI engine, the current codebase has being migrated and will continue for some time, so it's possible to find Blaze templates still and some changes or contributions may need to be made in Blaze while we are still evolving our components library.
 
-## Coding standards
+[Fuselage](https://github.com/RocketChat/Rocket.Chat.Fuselage) is our component library based on React, check it out when contributing to the Rocket.Chat UI and contribute new components or fixes for those components there.
 
-Most of the coding standards are covered by `.editorconfig` and `.eslintrc.js`.
+### Standards
+
+Most of the coding standards are covered by ESLint configured at [.eslintrc](../.eslintrc), and most os them came from our own [ESLint Config Package](https://github.com/RocketChat/eslint-config-rocketchat).
 
 Things not covered by `eslint`:
 
-* `exports`/`module.exports` should be at the end of the file
-* Longer, descriptive variable names are preferred, e.g. `error` vs `err`
+* Prefer longer/descriptive variable names, e.g. `error` vs `err`, unless dealing with common record properties already shortened, e.g. `rid` and `uid`
+* Use return early pattern. [See more](https://blog.timoxley.com/post/47041269194/avoid-else-return-early)
+* Prefer `Promise` over `callbacks`
+* Prefer `await` over `then/catch`
+* Don't create queries outside models, the query description should be inside the model class.
+* Don't hardcode fields inside models. Same method can be used for different purposes, using different fields.
+* Prefer create REST endpoints over Meteor methods
+* Prefer call REST endpoints over Meteor methods when both are available
+* v1 REST endpoints should follow the following pattern: `/api/v1/dashed-namespace.camelCaseAction`
+* Prefer TypeScript over JavaScript. Check [ECMAScript vs TypeScript](#ecmascript-vs-typescript)
 
-We acknowledge all the code does not meet these standards but we are working to change this over time.
+#### Blaze
+* Import the HTML file from it's sibling JS/TS file
 
 ### Syntax check
 
 Before submitting a PR you should get no errors on `eslint`.
 
-To check your files, first install `eslint`:
+To check your files run:
+
+```shell
+meteor npm run lint
+```
+
+## Tests
+
+There are 2 types of tests we run on Rocket.Chat, **Unit** tests and **End to End** tests. Different from the Unit tests the End to End depends on a Rocket.Chat instance running to execute the API and UI checks.
```

Comment:
Suggested change ...
"Different from the Unit tests the End to End depends on ..."
to
"The major difference is that End to End tests require ..."